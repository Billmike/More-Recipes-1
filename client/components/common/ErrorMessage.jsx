import React from 'react';
import PropTypes from 'prop-types';

const ErrorMessage = (props) => {
  const { message } = props;
  return (
    <div className="red-text center-align">
      {message}
    </div>
  );
};

ErrorMessage.propTypes = {
  message: PropTypes.string.isRequired,
};

export default ErrorMessage;
