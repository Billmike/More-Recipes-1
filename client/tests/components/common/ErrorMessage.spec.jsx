import React from 'react';

import ErrorMessage from '../../../components/common/ErrorMessage';

describe('Error message component', () => {
  it('should render component and display error message', () => {
    const props = {
      message: 'Username already exist',
    };
    const wrapper = shallow(<ErrorMessage {...props} />);
    expect(wrapper).toBeDefined();
    expect(wrapper.find('div').length).toEqual(1);
    expect(wrapper.find('div').text()).toEqual('Username already exist');
    expect(wrapper).toMatchSnapshot();
  });
  it('should display custom error message for unauthenticated user', () => {
    const props = {
      message: 'jwt expired',
    };
    const wrapper = shallow(<ErrorMessage {...props} />);
    expect(wrapper.find('div').text()).toEqual('Please login to continue');
    expect(wrapper).toMatchSnapshot();
  });
});
