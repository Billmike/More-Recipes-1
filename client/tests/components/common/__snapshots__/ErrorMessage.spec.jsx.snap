// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error message component should display custom error message for unauthenticated user 1`] = `
<div
  className="red-text center-align"
  id="show-error"
>
  Please login to continue
</div>
`;

exports[`Error message component should render component and display error message 1`] = `
<div
  className="red-text center-align"
  id="show-error"
>
  Username already exist
</div>
`;
